File: app/models/studio.rb
class Studio
  class << self
    def find(lat, lng)
      url = "http://yoganow-api.herokuapp.com/api/yoga_studios.json?lat=#{lat}&lng=#{lng}"
      studios = HTTParty.get(url).parsed_response['studios']
    end

    def search_near_coordinates(lat, lng)
      url = "http://yoganow-api.herokuapp.com/api/yoga_studios.json?lat=#{lat}&lng=#{lng}"
      studios = HTTParty.get(url).parsed_response['studios']
    end

    def classes(studio_id)
      url = "http://yoganow-api.herokuapp.com/api/yoga_studios/#{studio_id}/yoga_classes.json"
      studio_and_classes = HTTParty.get(url).parsed_response
    end
  end 
end

File: public/scripts/models/studio.js
define(['backbone'], function (Backbone) {

  var Studio = Backbone.Model.extend({
    select: function () {
      console.log('studio selected');
    },
  
    createYogaClass: function (params, callback) {
      var self = this;
      params.yoga_studio_id = self.get('id');
      var url = 'http://yoganow-api.herokuapp.com/api/yoga_studios/'+self.get('id')+'/yoga_classes.json'
      $.ajax({
        type: 'POST',
        url: url,
        data: {
          yoga_class: params
        },
        success: function(data){
          callback(data);
        }
      })
    }
  });

  return Studio;
});